integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/loading/AbstractSearchQueryEntityLoadingIT.java

```diff
@@ -62,21 +71,45 @@
 
 			List<T> loadedEntities = query.fetchHits();
 
-			// Be sure to do this after having executed the query, to avoid polluting the persistence context
+			softAssertions.assertThat( loadedEntities )
+					.as(
+							"Loaded entities when targeting types " + targetClasses
+									+ " and when the backend returns document references " + hitDocumentReferences
+					)
+					.allSatisfy( loadedEntity -> {
+						// Loading should fully initialize entities
+						assertThat( Hibernate.isInitialized( loadedEntity ) ).isTrue();
+					} );
+
+			assertionsContributor.accept( softAssertions );
+
+			// Be sure to do this after having executed the query and checked loading,
+			// because it may trigger additional loading.
 			EntityCollector<T> entityCollector = new EntityCollector<>( session );
 			expectedLoadedEntitiesContributor.accept( entityCollector );
 			List<T> expectedLoadedEntities = entityCollector.collected;
 
-			softAssertions.assertThat( loadedEntities )
+			// Both the expected and actual list may contain proxies: unproxy everything so that equals() works correctly
+			List<T> unproxyfiedExpectedLoadedEntities = unproxyAll( expectedLoadedEntities );
+			List<T> unproxyfiedLoadedEntities = unproxyAll( loadedEntities );
+
+			softAssertions.assertThat( unproxyfiedLoadedEntities )
 					.as(
-							"Loaded entities when targeting types " + targetClasses
+							"Loaded, then unproxified entities when targeting types " + targetClasses
 									+ " and when the backend returns document references " + hitDocumentReferences
 					)
-					.containsExactlyElementsOf( expectedLoadedEntities );
-			assertionsContributor.accept( softAssertions );
+					.containsExactlyElementsOf( unproxyfiedExpectedLoadedEntities );
```

Comment:
We always want to load fully initialized entities?

Reply:
That's not what I'm checking. I'm just checking the identity of returned entities.

Quoting above:

> Both the expected and actual list may contain proxies: unproxy everything so that equals() works correctly

The returned objects may not be fully initialized, I don't check that since I initialize them myself in this test. Though in practice, I think they are always initialized.